### DISCUSSION TITLE
FinalizeAddAuthenticator call failure while setting up 2FA

### DISCUSSION THREAD

**Discussion by renitens** [dyc3/steamguard-cli#425] [Q&A] [answered]
Hi Dyc3 & Contributors!

I'm evaluating steamguard-cli to be used for setting up computers in a PC-cafe using a commercial steam account.
Platform: Windows 11 22H2 with latest updates, Powershell 5.1

Sequence of events:
1. Initially the account is set to use email 2FA. 
2. Entering user/pass goes fine.
3. Entering the email 2FA goes fine.
4. Login successful, revocation code is received.
5. When attempting the next step, after entering the new code to finalize linking I get a transport error to 
`https://api.steampowered.com/ITwoFactorService/FinalizeAddAuthenticator/v1`

Log of `.\steamguard.exe -v debug setup`:
~~steamguard-cli.txt~~ (removed, contained sensitive info)

Other information:
- I was logged in to steam client on the machine while setup was executed. I feared if something goes wrong, we would be locked out of the account, and did not want to contact the awesome Steam support unless absolutely necessary.
- I tried the flow 3 times total with some time between, the same thing happened. I could revert to email 2FA after the linking failed using the revocation code. Steam showed an error message saying the code was not correct, but the reversion happened regardless every time.

I hope you guys have an idea how to go forward^^ thanks!

**Comment by dyc3** [comment, marked as answer]
The error you posted is a network error, and the operation timed out. That means that Valve's servers didn't respond in time to the finalization. This is not a bug with steamguard-cli, Valve's `ITwoFactorService` is just kinda unreliable and enigmatic.

I usually recommend waiting anywhere between 2-24 hours for cases like these. I just tried `steamguard setup` and it worked fine for me.

**Reply by renitens** [reply]
Thanks for the quick reply! I'll try again later.
